projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	for {
+		resp, err := http.Get("http://localhost:8080")
+		if err != nil {
+			fmt.Println("Server not working at the moment!")
+			fmt.Fprint(os.Stderr, "Server not working at the moment! ", err, "\n")
+			return
+		}
+		defer resp.Body.Close()
```

Comment:
Using `defer` causes the function to get called at the end of the enclosing function call - in a loop that's generally not want you want (because it means you'll keep holding onto the resources the `defer` is meant to release across multiple loop iterations). It's common to extract a function for the contents of a loop you need to `defer` in to address this (and also, extracting that function will probably be useful for other reasons too, e.g. testing :))

Reply:
@illicitonion I can't understand about "you'll keep holding onto the resources the defer is meant to release across multiple loop iterations"
I deleted the "defer" and used it in different places.